packages/apollo-engine-reporting/src/agent.ts

```diff
@@ -16,6 +16,11 @@ import { GraphQLRequestContext } from 'apollo-server-types';
 import { InMemoryLRUCache } from 'apollo-server-caching';
 import { defaultEngineReportingSignature } from 'apollo-graphql';
 
+// TDOD: Reviewer, should we consider moving these exports to a different package
```

Comment:
I do think they should move somewhere else since they're used both in managed configuration of the gateway **and** the aer module.

Reply:
FWIW: The way things are set up now, if metrics-reporting is false but managed federation is on, then the user will never see a deprecation message

Reply:
(which is ... maybe not possible?)

Reply:
I'd go with `apollo-server-env` for now unless I can think something better. I think it's reasonable. Don't forget to cleanup the `apollo-engine-reporting` dependency you added to `apollo-server-cloud-functions` :)

Reply:
What about [`apollo-graphql`](https://github.com/apollographql/apollo-tooling/blob/4336cd27480a3bbf363e63ea179b1e234d017812/packages/apollo-graphql/package.json#L2), which is depended on by CLI, AS Core, AER, Gateway and federation?